libs/tlv/src/coder.rs

```diff
@@ -0,0 +1,165 @@
+use std::io::{Read, Write};
+
+use crate::length::Length;
+
+/// Encodes a value into a writer.
+pub trait Encode {
+    /// Converts the value into a byte representation and writes it to the
+    /// writer.
+    fn encode<W>(&self, writer: &mut W) -> std::io::Result<()>
```

Comment:
I went with a fairly uninformative `std::io::Result` for these rather than a more specific `Error`, but if I were to improve this library that's likely something I'd do to include things like field names, types, offsets, etc. a la `serde`.

Reply:
This has been fixed.